controllers/cloudstackmachinestatechecker_controller.go

```diff
@@ -81,21 +81,23 @@ func (r *CloudStackMachineStateCheckerReconciliationRunner) Reconcile() (ctrl.Re
 
 	csRunning := r.CSMachine.Status.InstanceState == "Running"
 	csTimeInState := r.CSMachine.Status.TimeSinceLastStateChange()
-	capiRunning := r.CAPIMachine.Status.Phase == "Running"
+	capiPhase := capiv1.MachinePhase(r.CAPIMachine.Status.Phase)
+	capiRunning := capiPhase == capiv1.MachinePhaseRunning
+	deleteInvoked := capiPhase == capiv1.MachinePhaseDeleting || capiPhase == capiv1.MachinePhaseDeleted
```

Comment:
I thought the main issue we were trying to address was deleting CAPI machines on move operations in a slow Cloudstack environment? I don't see that logic being implemented in this PR. Do we have a plan to address it in a follow-up PR?

Reply:
That's an unrelated issue to be addresses separately.

Reply:
That's an unrelated issue to be addresses separately.